src/pages/index.js

```diff
@@ -83,87 +82,109 @@ function handleDeleteIcon(cardId, card) {
     api
       .deleteCard(cardId)
       .then(() => {
-        deleteCardPopup.close();
         card.remove();
       })
       .catch((err) => {
         console.error(err);
+      })
+      .finally(() => {
+        deleteCardPopup.close();
       });
   });
 }
 
 function handleProfileImgFormSubmit(data) {
   api
     .updateProfileImg(data.link)
+    .then(profileImagePopup.editSubmitBtn())
     .then(() => {
       userInfo.setUserProfileImage(data.link);
-      profileImagePopup.close();
     })
-    .catch((error) => {
-      console.error("Error fetching data:", error);
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      profileImagePopup.resetSubmitBtn();
+      profileImagePopup.close();
     });
 }
 
 function handleCardClick(name, link) {
   previewImagePopup.open(name, link);
 }
-function addLike(cardId, likeCounter) {
+function addLike(cardId, likeCounter, likeBtn) {
   api
     .addLike(cardId)
     .then((res) => {
       likeCounter.textContent = res.likes.length;
+      console.log(likeBtn);
     })
-    .catch((error) => {
-      console.error("Error fetching data:", error);
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      likeBtn.classList.add("card__like-button_active");
```

Comment:
`"card__like-button_active"`

Now only class `Card` knows every card selector, so you need to encapsulate the likes logics inside its methods and call them in `index.js`

Reply:
I'm not sure I understand. Are you saying I should should be doing the classlist changes on the like icon inside of the Card class instead of writing them in index.js?

Reply:
> I'm not sure I understand. Are you saying I should should be doing the classlist changes on the like icon inside of the Card class instead of writing them in index.js?

Yes, that's right


You've done it correctly